components/eamxx/src/diagnostics/potential_temperature.cpp

```diff
@@ -7,7 +7,26 @@ namespace scream
 PotentialTemperatureDiagnostic::PotentialTemperatureDiagnostic (const ekat::Comm& comm, const ekat::ParameterList& params)
   : AtmosphereDiagnostic(comm,params)
 {
-  // Nothing to do here
+  EKAT_REQUIRE_MSG(params.isParameter("Temperature Kind"),
+      "Error! PotentialTemperatureDiagnostic requires 'Temperature Kind' in its input parameters.\n");
+  
+  auto pt_type = params.get<std::string>("Temperature Kind");
+
+  if (pt_type=="Base"){
```

Comment:
Could we avoid string comparisons? May it be that shoc is using this function in perf part?